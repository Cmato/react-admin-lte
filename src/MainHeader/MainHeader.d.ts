import * as React from 'react';

export interface MainHeaderProps {
  component?: React.ReactType<MainHeaderProps>;
  largeLogo?: React.ReactNode;
  logoComponent?: ((props: {
    className: string;
    children: React.ReactNode
  }) => React.ReactNode) | React.ReactNode;
  logoHref?: string;
  miniLogo?: React.ReactNode;
  onSidebarCollapseClick?: React.MouseEventHandler<any>;
}

declare const MainHeader: React.ComponentType<MainHeaderProps>;

export default MainHeader;
