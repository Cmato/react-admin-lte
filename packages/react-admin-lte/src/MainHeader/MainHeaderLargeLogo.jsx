import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';

const propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

const MainHeaderLargeLogo = ({
  children,
  className,
}) => {
  const classes = {
    'logo-lg': true,
  };

  return (
    <span className={classNames(className, classes)}>{children}</span>
  );
};

MainHeaderLargeLogo.propTypes = propTypes;

export default MainHeaderLargeLogo;
