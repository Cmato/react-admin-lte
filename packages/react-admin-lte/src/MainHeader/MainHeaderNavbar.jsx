import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';

const propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

const MainHeaderNavbar = ({
  children,
  className,
}) => {
  const classes = {
    navbar: true,
    'navbar-static-top': true,
  };

  return (
    <nav className={classNames(className, classes)}>
      {children}
    </nav>
  );
};

MainHeaderNavbar.propTypes = propTypes;

export default MainHeaderNavbar;
