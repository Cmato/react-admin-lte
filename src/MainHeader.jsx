import React from 'react';

import MainHeaderLogo from './MainHeaderLogo';

const propTypes = {
  children: React.PropTypes.node,
};

const MainHeader = ({ children }) => (
  <header className="main-header">
    {children}
  </header>
);

MainHeader.propTypes = propTypes;

MainHeader.Logo = MainHeaderLogo;

export default MainHeader;
