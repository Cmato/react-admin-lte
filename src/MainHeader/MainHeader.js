import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const MainHeader = ({
  children,
  className: classNameProp,
  component: Component,
  largeLogo,
  logoComponent,
  logoHref,
  miniLogo,
  onSidebarCollapseClick,
  ...other
}, context) => {
  const className = classNames({
    'main-header': true
  }, classNameProp);

  let handleCollapseClick;

  if (onSidebarCollapseClick) {
    handleCollapseClick = onSidebarCollapseClick;
  } else if (context.$adminlte_theme) {
    handleCollapseClick = (e) => { context.$adminlte_theme.onChange(e) };
  }

  const logos = [
    <span key="mini" className="logo-mini">{miniLogo}</span>,
    <span key="large" className="logo-lg">{largeLogo}</span>
  ];

  const logo = typeof logoComponent === 'function'
    ? logoComponent({ className: 'logo', children: logos })
    : <a className="logo" href={logoHref}>{logos}</a>;

  return (
    <Component className={className} {...other}>
      {logo}
      <div className="navbar navbar-static-top">
        <a
          className="sidebar-toggle"
          onClick={handleCollapseClick}
          role="button"
        >
          <span className="sr-only">Toggle navigation</span>
        </a>
        <div className="navbar-custom-menu">
          {children}
        </div>
      </div>
    </Component>
  );
};

MainHeader.propTypes = {
  /**
   * The contents of the main footer.
   */
  children: PropTypes.node,
  /**
   * @ignore
   */
  className: PropTypes.string,
  /**
   * The component used for the root node.
   * Either a string to use a DOM element or a component.
   */
  component: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
  /**
   * The logo to display when sidebar is expanded.
   */
  largeLogo: PropTypes.node,
  /**
   * Override rendering of logo. If not supplied is `a` by default.
   */
  logoComponent: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
  /**
   * Supplied to logo `a` element if not overriden.
   */
  logoHref: PropTypes.string,
  /**
   * The logo to display when mini sidebar is collapsed.
   */
  miniLogo: PropTypes.node,
  /**
   * Callback fired when the sidebar collapse is clicked.
   */
  onSidebarCollapseClick: PropTypes.func
};

MainHeader.defaultProps = {
  component: 'header'
};

MainHeader.contextTypes = {
  $adminlte_theme: PropTypes.shape({
    onChange: PropTypes.func
  })
};

export default MainHeader;
