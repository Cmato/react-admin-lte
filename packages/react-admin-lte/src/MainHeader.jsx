import PropTypes from 'prop-types';
import React from 'react';

import MainHeaderLogo from './MainHeaderLogo';
import MainHeaderSidebarToggle from './MainHeaderSidebarToggle';
import MainHeaderNavbar from './MainHeaderNavbar';

const propTypes = {
  children: PropTypes.node,
};

const MainHeader = ({ children }) => (
  <header className="main-header">
    {children}
  </header>
);

MainHeader.propTypes = propTypes;

MainHeader.Logo = MainHeaderLogo;
MainHeader.SidebarToggle = MainHeaderSidebarToggle;
MainHeader.Navbar = MainHeaderNavbar;

export default MainHeader;
