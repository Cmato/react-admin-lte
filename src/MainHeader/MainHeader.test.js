import React, {Fragment} from 'react';
import { shallow } from 'enzyme';
import MainHeader from './MainHeader';

it('renders default', () => {
  expect(shallow(<MainHeader />))
    .toMatchSnapshot();
});

it('renders children', () => {
  expect(shallow(<MainHeader><div /></MainHeader>))
    .toMatchSnapshot();
});

it('renders custom component', () => {
  expect(shallow(<MainHeader component="div" />))
    .toMatchSnapshot();
});

it('renders mini logo', () => {
  expect(shallow(<MainHeader miniLogo={<Fragment><b>A</b>LT</Fragment>} />))
    .toMatchSnapshot();
});

it('renders large logo', () => {
  expect(shallow((
    <MainHeader largeLogo={<Fragment><b>Admin</b>LTE</Fragment>} />
  ))).toMatchSnapshot();
});

it('renders logo href', () => {
  expect(shallow((
    <MainHeader logoHref="/" />
  ))).toMatchSnapshot();
});

it('renders custom logo', () => {
  expect(shallow((
    <MainHeader
      logoComponent={({ className, children }) => {
        return (
          <div className={className}>
            {children}
          </div>
        );
      }}
    />
  ))).toMatchSnapshot();
});

it('toggle sidebar through context', () => {
  let received = false;

  const wrapper = shallow(
    <MainHeader />,
    {
      context: {
        $adminlte_theme: {
          onChange: () => {
            received = true;
          }
        }
      }
    }
  );

  wrapper.find('.sidebar-toggle').simulate('click');
  expect(received).toEqual(true);
});

it('wires up onSidebarCollapseClick instead of context', () => {
  let receivedContext = false;
  let receivedClick = false;

  const wrapper = shallow(
    <MainHeader
      onSidebarCollapseClick={() => { receivedClick = true; }}
    />,
    {
      context: {
        $adminlte_theme: {
          onChange: () => {
            receivedContext = true;
          }
        }
      }
    }
  );

  wrapper.find('.sidebar-toggle').simulate('click');
  expect(receivedContext).toEqual(false);
  expect(receivedClick).toEqual(true);
});
