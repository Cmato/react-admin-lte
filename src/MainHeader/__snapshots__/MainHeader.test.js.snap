// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders children 1`] = `
<header
  className="main-header"
>
  <a
    className="logo"
  >
    <span
      className="logo-mini"
      key="mini"
    />
    <span
      className="logo-lg"
      key="large"
    />
  </a>
  <div
    className="navbar navbar-static-top"
  >
    <a
      className="sidebar-toggle"
      role="button"
    >
      <span
        className="sr-only"
      >
        Toggle navigation
      </span>
    </a>
    <div
      className="navbar-custom-menu"
    >
      <div />
    </div>
  </div>
</header>
`;

exports[`renders custom component 1`] = `
<div
  className="main-header"
>
  <a
    className="logo"
  >
    <span
      className="logo-mini"
      key="mini"
    />
    <span
      className="logo-lg"
      key="large"
    />
  </a>
  <div
    className="navbar navbar-static-top"
  >
    <a
      className="sidebar-toggle"
      role="button"
    >
      <span
        className="sr-only"
      >
        Toggle navigation
      </span>
    </a>
    <div
      className="navbar-custom-menu"
    />
  </div>
</div>
`;

exports[`renders custom logo 1`] = `
<header
  className="main-header"
>
  <div
    className="logo"
  >
    <span
      className="logo-mini"
      key="mini"
    />
    <span
      className="logo-lg"
      key="large"
    />
  </div>
  <div
    className="navbar navbar-static-top"
  >
    <a
      className="sidebar-toggle"
      role="button"
    >
      <span
        className="sr-only"
      >
        Toggle navigation
      </span>
    </a>
    <div
      className="navbar-custom-menu"
    />
  </div>
</header>
`;

exports[`renders default 1`] = `
<header
  className="main-header"
>
  <a
    className="logo"
  >
    <span
      className="logo-mini"
      key="mini"
    />
    <span
      className="logo-lg"
      key="large"
    />
  </a>
  <div
    className="navbar navbar-static-top"
  >
    <a
      className="sidebar-toggle"
      role="button"
    >
      <span
        className="sr-only"
      >
        Toggle navigation
      </span>
    </a>
    <div
      className="navbar-custom-menu"
    />
  </div>
</header>
`;

exports[`renders large logo 1`] = `
<header
  className="main-header"
>
  <a
    className="logo"
  >
    <span
      className="logo-mini"
      key="mini"
    />
    <span
      className="logo-lg"
      key="large"
    >
      <React.Fragment>
        <b>
          Admin
        </b>
        LTE
      </React.Fragment>
    </span>
  </a>
  <div
    className="navbar navbar-static-top"
  >
    <a
      className="sidebar-toggle"
      role="button"
    >
      <span
        className="sr-only"
      >
        Toggle navigation
      </span>
    </a>
    <div
      className="navbar-custom-menu"
    />
  </div>
</header>
`;

exports[`renders logo href 1`] = `
<header
  className="main-header"
>
  <a
    className="logo"
    href="/"
  >
    <span
      className="logo-mini"
      key="mini"
    />
    <span
      className="logo-lg"
      key="large"
    />
  </a>
  <div
    className="navbar navbar-static-top"
  >
    <a
      className="sidebar-toggle"
      role="button"
    >
      <span
        className="sr-only"
      >
        Toggle navigation
      </span>
    </a>
    <div
      className="navbar-custom-menu"
    />
  </div>
</header>
`;

exports[`renders mini logo 1`] = `
<header
  className="main-header"
>
  <a
    className="logo"
  >
    <span
      className="logo-mini"
      key="mini"
    >
      <React.Fragment>
        <b>
          A
        </b>
        LT
      </React.Fragment>
    </span>
    <span
      className="logo-lg"
      key="large"
    />
  </a>
  <div
    className="navbar navbar-static-top"
  >
    <a
      className="sidebar-toggle"
      role="button"
    >
      <span
        className="sr-only"
      >
        Toggle navigation
      </span>
    </a>
    <div
      className="navbar-custom-menu"
    />
  </div>
</header>
`;
